Add docker config and setup for local development

luizreis opened the pull request:
Fixes #1175.

#### Changes proposed in this Pull Request:

- Add docker config and setup
- Add Xdebug config using the environment variables in `default.env`

#### Testing instructions:

1. Run `npm run up` to create the containers
2. After the containers are created, run `./bin/docker-setup.sh`
3. Navigate to http://localhost:8082 and check if the site is configured with sample products and Stripe is activated
4. Test Xdebug by listening on port 9000 and mapping `"/var/www/html/wp-content/plugins/woocommerce-gateway-stripe"` to the workspace folder
5. Run `npm run down` to stop the containers

dwainm commented:
For simplicity's sake, we should just have everything behind npm. I suggest we add `npm run docker-init` to run the docker-setup.sh.

Would it be good to have a command that will run both `docker-init` and `up`?

dwainm commented on tests/README.md:
````diff
@@ -11,7 +11,7 @@
 	Pass db accesss arguments as params e.g. `tests/bin/install.sh db_name user password db_host`.
 	Example command if running DB in docker and port 5678 is exposed by docker container:
 	```
-	tests/bin/install.sh test_gateway root worpdress 127.0.0.1:5678
````
good catch :)

luizreis commented:
> For simplicity's sake, we should just have everything behind npm. I suggest we add npm run docker-init to run the docker-setup.sh.

Is there any value in adding an npm script just for this? The setup is only run a single time when setting up the containers. If so, should we add a script to run `tests/bin/install.sh` as well?

> Would it be good to have a command that will run both docker-init and up?

I guess we could add one to run the first time docker setup depending on the answer above, instead of just running `docker-setup.sh`. Something like `docker-compose up -d && bin/docker-setup.sh && tests/bin/install.sh ...default_docker_args...`

> One change: tests/bin/install.sh should probably remain in 0644.

May I ask why? I've taken a look at our other repositories and executables seem to be set to `0755`. I changed it due to getting the following error when trying to run it using the commands from our instructions: `permission denied: tests/bin/install.sh`.

> Things I can see is adding is a command runner: npm run wp ...cli_command_here...

I like this suggestion, it'd make it much easier than `docker run -it --rm --user xfs --volumes-from woocommerce_stripe_wordpress --network container:woocommerce_stripe_wordpress wordpress:cli` 😆. Added in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1188/commits/761bbe55182f9a54ef45732a7fc703cb05623fd7.

I also think it would be nice to prefix the docker commands, especially as we add more of them. What do you think? I did it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1188/commits/e97a02d35ab773bcc7423e49be415cb19bc1943b, but we can revert it if you don't agree.

luizreis commented:
> I'd prefer WP cli to not be docker CLI. It is nice if it is called WP.

@dwainm - yeah, I was a little unsure of the name for that. My initial thought was `wp` as well, however we would end up with commands like `npm run docker:wp wp ...` since we probably need to support `npm run docker:wp config ...` as well. Do you still think we should use wp?
